web/blog/tenzir-node-v4.28/index.md

```diff
@@ -0,0 +1,301 @@
+      sin: 898,
+      sout: 819,
+      sport: 80,
+      status: "OK",
+      url: "http://test.de/images/icons/144x144.png",
+    },
+  },
+}
+```
+
+Let's now reshape the data, keeping only the fields we're interested in:
+
+```tql
+load_file "sample.txt"
+read_syslog
+content = content.parse_grok("%{DATA:id} %{DATA:ts} %{DATA:action} %{GREEDYDATA:params}")
+content.params = content.params.parse_kv()
+// Interpret the timestamp as a seconds since Unix epoch.
+content.ts = from_epoch(content.ts * 1s)
+// Interpret the duration parameter as seconds.
+content.params.duration = content.params.duration * 1s
+// Split the forwarded_for field into an array of IP addresses.
+content.params.forwarded_for = content.params.forwarded_for.split(", ").map(x, x.ip())
+this = {
+  // We keep both timestamps, labelling the one from the message `time` and the
+  // one from Syslog itself `forwarded_time`:
+  time: content.ts,
+  forwarded_time: timestamp,
+  // Copy the other id and the action from the Syslog message:
+  id: content.id,
+  action: content.action,
+  // Lastly, take all the params:
+  ...content.params,
+}
+```
+
+```tql
+{
+  time: 2023-12-21T18:18:16Z,
+  forwarded_time: 2023-06-09T05:51:36.433Z,
+  id: "I103",
+  action: "request",
+  baddr: 194.1.1.1,
+  bport: 58630,
+  c_ctype: "image/png",
+  caddr: 194.1.1.1,
+  cin: 860,
+  cout: 898,
+  cport: 58630,
+  duration: 0ns,
+  forwarded_for: [
+    192.1.1.1,
+    191.1.1.1,
+  ],
+  laddr: 127.128.11.145,
+  lport: 1,
+  method: "GET",
+  paddr: 172.0.0.1,
+  pport: 80,
+  proto: 6,
+  relay_name: "wwwserver",
+  response_code: 200,
+  rnum: 6489895,
+  s_ctype: "image/png",
+  saddr: 172.0.0.1,
+  sin: 898,
+  sout: 819,
+  sport: 80,
+  status: "OK",
+  url: "http://test.de/images/icons/144x144.png",
+}
+```
+
+Now we just need to switch from reading from a file to accepting Syslog over UDP
+directly, and we're done. That's as simple as replacing `load_file "…"` with
+`load_udp "…"` or `load_tcp "…"`.
+
+Besides the `parse_kv` and `parse_grok` functions shown here, we also introduced
+other new functions in this release for parsing nested data:
+
+- `parse_csv`, `parse_ssv`, and `parse_tsv` parse comma-, space-, and
+  tab-separated values, respectively.
+- `parse_leef` and `parse_cef` parse LEEF and CEF data, respectively.
+- `parse_syslog` parses Syslog data.
+- `parse_json` parses JSON values.
+- `parse_yaml` parses YAML values.
+
+## VSCode TQL Package
+
+If you're writing TQL in Visual Studio Code, you can now enjoy syntax
+highlighting just like in the Tenzir Platform: The VSCode TQL package is now
+available on the [Visual Studio Marketplace][vscode-tql].
+
+[vscode-tql]: https://marketplace.visualstudio.com/items?itemName=tenzir.vscode-tql
+
+## Nodes are now Faster
+
+We've made a few optimizations to improve the performance of Tenzir Nodes.
+
+In particular, we've optimized the following:
+
+- The `read_ndjson` and `write_ndjson` operator are now faster.
```

Comment:
This needs a quantifier. The reader is left with "by how much"?

Reply:
The section is pretty weak generally. The numbers are hard to quantify, since they depend on many factors. I'm considering just removing the entire paragraph.

Reply:
If we cannot quantify it, the we should indeed remove it.